ARCs/arc-0047.md

```diff
@@ -0,0 +1,169 @@
+
+### Interface
+
+```typescript
+interface LogicSignatureDescription {
+    name: string,
+    description: string,
+    program: string,
+    variables: {
+        variable: string,
+        name: string,
+        type: string,
+        description: string
+    }[]
+}
+```
+
+| Key | Description |
+| --- | ----------- |
+| `name` | The name of the logic signature. **SHOULD** be short and descriptive |
+| `description` | A description of what the logic signature does |
+| `program` | base64 encoding of the TEAL program source
+| `variables` | An array of variables in the program |
+| `variables.variable` | The name of the variable in the templated program. |
+| `variables.name` | Human-friendly name for the variable. **SHOULD** be short and descriptive |
+| `variables.type` | **MUST** be a type defined below in the `type` section |
+| `variables.description` | A description of how this variable is used in the program |
+
+### Variables
+
+A variable in the program **MUST** be start with `TMPL_`
+
+#### Types
+
+Any valid ABI type encoded in base16 (without the leading `0x`) with the following exceptions:
+
+| Type | Description |
+| ---- | ----------- |
+| `address` | 58-character Algorand public address. Typically to be used as an argument to the `addr` opcode. Front-ends **SHOULD** provide a link to the address on an explorer
+| `application` | Application ID. Alias for `uint64`. Front-ends **SHOULD** provide a link to the app on an explorer |
+| `asset` | Asset ID. Alias for `uint64`. Front-ends **SHOULD** provide a link to the asset on an explorer |
+| `string` | ASCII string. Typically used as an argument to `byte`, `method`, or a branching opcode.
+| `hex` | base16 encoding of binary data. Typically used as an argument to `byte`. **MUST** be prefixed with `0x` |
+
+For all other types, front-ends **MUST** decode the ABI type to display the human-readable value to the user. 
+
+### Input Validation
+
+All ABI types **MUST** be encoded as base16 and prefixed with `0x`, with the exception of `uint64` which should be provided as an integer.
+
+String values **MUST NOT** include any unescpaed `"` to ensure there is no TEAL injection.
+
+All values **MUST** be validated to ensure they are encoded properly. This includes the following checks:
+
+* An `address` value must be a valid Algorand address
+* A `uint64`, `application`, or `asset` value must be a valid unsigned 64-bit integer
+
+### Unique Identification
+
+To enable unique identification of a description, clients **MUST** calculate the SHA256 hash of the JSON description canonicalized in accordance with RFC 8785.
+
+## Rationale
+
+This provides a way for frontends to clearly display to the user what is being signed when signing a logic signature. 
+
+Template variables must be  an immediate argument otherwise a string variable could specify the opcode in the program, which could have unintended and unclear consequences. 
```

Comment:
```suggestion
Template variables must be immediate arguments. Otherwise a string variable could specify an opcode in the program, which could have unintended and unclear consequences.
```

This rule implies wallets can parse TEAL. That's asking a lot.